test/e2e/dns_autoscaling.go

```diff
@@ -38,42 +38,60 @@ const (
 	DNSdefaultTimeout      = 5 * time.Minute
 	DNSNamespace           = "kube-system"
 	ClusterAddonLabelKey   = "k8s-app"
-	KubeDNSLabelName       = "kube-dns"
+	DNSLabelName           = "kube-dns"
 	DNSAutoscalerLabelName = "kube-dns-autoscaler"
 )
 
 var _ = framework.KubeDescribe("DNS horizontal autoscaling", func() {
 	f := framework.NewDefaultFramework("dns-autoscaling")
 	var c clientset.Interface
 	var previousParams map[string]string
-	DNSParams_1 := DNSParamsLinear{map[string]string{"linear": "{\"nodesPerReplica\": 1}"}, 1.0, 0.0}
-	DNSParams_2 := DNSParamsLinear{map[string]string{"linear": "{\"nodesPerReplica\": 2}"}, 2.0, 0.0}
-	DNSParams_3 := DNSParamsLinear{map[string]string{"linear": "{\"nodesPerReplica\": 3, \"coresPerReplica\": 3}"}, 3.0, 3.0}
+	var originDNSReplicasCount int
+
+	DNSParams_1 := DNSParamsLinear{
+		nodesPerReplica: 1,
+	}
+	DNSParams_2 := DNSParamsLinear{
+		nodesPerReplica: 2,
+	}
+	DNSParams_3 := DNSParamsLinear{
+		nodesPerReplica: 3,
+		coresPerReplica: 3,
+	}
 
 	BeforeEach(func() {
 		framework.SkipUnlessProviderIs("gce")
 		c = f.ClientSet
 
 		Expect(len(framework.GetReadySchedulableNodesOrDie(c).Items)).NotTo(BeZero())
 
+		By("Collecting original replicas count and DNS scaling params")
+		var err error
+		originDNSReplicasCount, err = getDNSReplicas(c)
```

Comment:
Sorry, I checked the code and it turns out we check all nodes are ready after every test([ref1](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/framework.go#L142) and [ref2](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/framework.go#L437)).

But I think it would be suffice for this case?